### PR TITLE
Multi currency prices

### LINKED ISSUE #1: Add currency app
It should contains `ConversionRate` model and `update_conversion_rates` util.

### PR DESCRIPTION
**Pull request by artursmet**
Fixes #1

### CODE REVIEW

File: django_prices_openexchangerates/models.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
+from decimal import Decimal, ROUND_HALF_UP
```
Comment (patrys) [inline django_prices_openexchangerates/models.py]: Missing import of unicode literals.

File: django_prices_openexchangerates/models.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+    def recalculate_currencies(self):
+        from .utils import convert_price
+        for currency in settings.AVAILABLE_PURCHASE_CURRENCIES:
+            if currency in CURRENCIES:
+                self.currencies[currency] = convert_price(self.price, currency)
+
+    @property
+    def in_base_currency(self):
+        return self.price
```
Comment (patrys) [inline django_prices_openexchangerates/models.py]: Please rename `self.price` to `self.base_price` and get rid of this property.

File: django_prices_openexchangerates/models.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            raise ValidationError(
+                'Can\'t set a conversion rate for the same currency')
+        super(ConversionRate, self).clean()
+
+    def __str__(self):
+        return '1 {} = {:.4f} {}'.format(
+            self.base_currency, self.rate, self.to_currency)
+
+
+class MultiCurrencyPrice(object):
```
Comment (patrys) [inline django_prices_openexchangerates/models.py]: Can we drop this entirely in favour of:

```python
exchange_currency(Price(10, currency='USD'), 'GBP')
```

File: django_prices_openexchangerates/models.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+        for currency in settings.AVAILABLE_PURCHASE_CURRENCIES:
+            if currency in CURRENCIES:
+                self.currencies[currency] = convert_price(self.price, currency)
+
+    @property
+    def in_base_currency(self):
+        return self.price
+
+
+class MultiCurrencyPriceField(PriceField):
```
Comment (patrys) [inline django_prices_openexchangerates/models.py]: This should not be needed. A template filter should solve the same use case.

File: django_prices_openexchangerates/tasks.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    return ExchangeRates(rates=exchange_data['rates'])
+
+
+def update_conversion_rates():
+    exchange_rates = get_latest_exchange_rates()
+    conversion_rates = ConversionRate.objects.all()
+    for conversion_rate in conversion_rates:
+        new_exchange_rate = exchange_rates[conversion_rate.to_currency]
+        conversion_rate.rate = new_exchange_rate
+        conversion_rate.save()
```
Comment (patrys) [inline django_prices_openexchangerates/tasks.py]: List fields to avoid overwriting all fields?

**Comment by artursmet** [comment]
@patrys As we discussed - now there is only one API, `exchange_currency()` method and a few template tags. Please review again.

File: django_prices_openexchangerates/models.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        try:
+            rate = _thread_locals.conversion_rates[to_currency]
+        except KeyError:
+            rate = self.get(to_currency=to_currency)
+            _thread_locals.conversion_rates[to_currency] = rate
+        return rate
+
+
+@python_2_unicode_compatible
+class ConversionRate(models.Model):
```
Comment (patrys) [inline django_prices_openexchangerates/models.py]: Please add a `__repr__` that includes the PK (and possibly currency and rate if that's not too much noise).

File: django_prices_openexchangerates/templatetags/currency_conversion.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+from django_prices.templatetags import prices_i18n
+
+from ..utils import exchange_currency
+
+
+register = Library()
+
+
+@register.simple_tag
+def gross(price, currency):
```
Comment (patrys) [inline django_prices_openexchangerates/templatetags/currency_conversion.py]: `gross_in_currency`?

File: django_prices_openexchangerates/templatetags/currency_conversion.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    converted_price = exchange_currency(price, currency)
+    return prices_i18n.gross(converted_price)
+
+
+@register.simple_tag
+def net(price, currency):
```
Comment (patrys) [inline django_prices_openexchangerates/templatetags/currency_conversion.py]: Ditto.

File: django_prices_openexchangerates/templatetags/currency_conversion.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    converted_price = exchange_currency(price, currency)
+    return prices_i18n.net(converted_price)
+
+
+@register.simple_tag
+def tax(price, currency):
```
Comment (patrys) [inline django_prices_openexchangerates/templatetags/currency_conversion.py]: And ditto.

File: django_prices_openexchangerates/tasks.py [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+
+BASE_URL = r'http://openexchangerates.org/api'
+ENDPOINT_LATEST = BASE_URL + r'/latest.json'
+try:
+    API_KEY = settings.OPENEXCHANGERATES_API_KEY
+except AttributeError:
+    raise ImproperlyConfigured('OPENEXCHANGERATES_API_KEY is required')
+
+
+class ExchangeRates(object):
```
Comment (patrys) [inline django_prices_openexchangerates/tasks.py]: As this only has one method, wouldn't it be more readable as a function?

```python
def extract_rate(rates, currency):
    base_rate = rates[settings.DEFAULT_CURRENCY]
    return rates[currency] / base_rate
```

File: django_prices_openexchangerates/templatetags/currency_conversion.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
+from django.template import Library
+from django_prices.templatetags import prices_i18n
+
+from ..utils import exchange_currency
+
+
```
Comment (patrys) [inline django_prices_openexchangerates/templatetags/currency_conversion.py]: Superfluous newline

File: django_prices_openexchangerates/utils.py [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+        try:
+            converted_price = rate.to_base_currency(converted_price)
+        except ZeroDivisionError:
+            logger.error(
+                'Conversion rate for %s is zero, failed to convert',
+                rate.to_currency)
+            raise ValueError('conversion rate for %s is 0', rate.to_currency)
+
+    return Price(currency=to_currency, net=converted_price,
+                 history=price.history)
```
Comment (patrys) [inline django_prices_openexchangerates/utils.py]: It would be nice to be able to see the currency switch in the history somehow.

**Comment by artursmet** [comment]
@patrys I added `PriceModifier` and I convert currencies there. I also refactored old `exchange_currency` function, now it's more readable.
Example with price modifier:

```python
In [3]: exchange_currency(Price(10, currency='PLN'), 'EUR').history
Out[3]: ((Price('10', currency='PLN') * CurrencyConversion(base_currency=PLN, to_currency=USD, rate=0.2670)) * CurrencyConversion(base_currency=USD, to_currency=EUR, rate=0.8840))
```

**Pull request merged**